src/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -385,10 +383,299 @@ public static void EnsureSocketsAreInitialized()
                             throw new SocketException((int)errorCode);
                         }
 
+                        s_getAddrInfoExSupported = TestGetAddrInfoEx();
+
                         Volatile.Write(ref s_initialized, true);
                     }
                 }
             }
         }
+
+        private static bool TestGetAddrInfoEx()
+        {
+            using (SafeLibraryHandle libHandle = Interop.Kernel32.LoadLibraryExW(Interop.Libraries.Ws2_32, IntPtr.Zero, 0))
+            {
+                return Interop.Kernel32.GetProcAddress(libHandle, nameof(Interop.Winsock.GetAddrInfoExCancel)) != IntPtr.Zero;
+            }
+        }
+
+        public static unsafe void GetAddrInfoAsync(string name, DnsResolveAsyncResult asyncResult)
+        {
+            try
+            { }
+            finally
+            {
```

Comment:
This try/finally pattern is not needed.  It's only relevant for thread aborts, which don't exist in .NET Core.

Reply:
Done